core/constant_inc.php

```diff
@@ -591,8 +592,11 @@
 define( 'DB_FIELD_SIZE_PASSWORD', 64 );
 define( 'DB_FIELD_SIZE_API_TOKEN_NAME', 128 );
 
-# Maximum size for the user's password when storing it as a hash
-define( 'PASSWORD_MAX_SIZE_BEFORE_HASH', 1024 );
+# Maximum size for the user's password when storing it as a hash.
+# The CRYPT_BLOWFISH algorithm used by password_hash() will silently truncate
+# passwords to 72 characters, so there is no point in allowing longer passwords.
+# @see http://php.net/function.crypt
+define( 'PASSWORD_MAX_SIZE_BEFORE_HASH', 72 );
```

Comment:
Is it worth adding complexity to the code to deal with a theoretical scenario that is unlikely to be an issue ? 

I mean, a password longer than 72 chars, _really_ ? `This is a really long password that I don't want to type everyday` is 66 chars...

An alternative could be to let BCRYPT silently truncate the password, the likelihood of a collision is